server/handler.go

```diff
@@ -103,6 +106,30 @@ func (h *Handler) ComQuery(
 
 	var r *sqltypes.Result
 	var proccesedAtLeastOneBatch bool
+
+	rowchan := make(chan sql.Row)
+	errchan := make(chan error)
+
+	// This goroutine will be selected giving a change to Vitess to call the
+	// handler.CloseConnection callback and enforcing the Timeout if configured
+	go func(rowc chan sql.Row, errc chan error) {
+		for {
+			row, err := rows.Next()
+			if err != nil {
+				errc <- err
+				return
+			}
+			rowc <- row
+		}
+	}(rowchan, errchan)
+
+	// Default (big) waitTime is 3600, but it wont matter if there is no timeout
+	// because the loop will just iterate again. If there is a timeout, it will
+	// be enforced to ensure that Vitess has a chance to call Handler.CloseConnection()
+	waitTime := 3600 * time.Second
```

Comment:
Could also be `waitTime := time.Hour`

Reply:
True. You can get a programmer out of Python but you can't get the Python out of the programmer, it seems.